platform/chunks_download_strategy.cpp

```diff
@@ -89,39 +90,42 @@ int64_t ChunksDownloadStrategy::LoadOrInitChunks(string const & fName, int64_t f
   ASSERT ( fileSize > 0, () );
   ASSERT ( chunkSize > 0, () );
 
-  try
+  if (Platform::IsFileExistsByFullPath(fName))
   {
-    FileReader r(fName);
-    ReaderSource<FileReader> src(r);
-
-    int64_t const readSize = ReadVarInt<int64_t>(src);
-    if (readSize == fileSize)
+    try
     {
-      // Load chunks.
-      uint64_t const size = src.Size();
-      int const stSize = sizeof(ChunkT);
-      size_t const count = size / stSize;
-      ASSERT_EQUAL(size, stSize * count, ());
-
-      m_chunks.resize(count);
-      src.Read(&m_chunks[0], stSize * count);
-
-      // Reset status "downloading" to "free".
-      int64_t downloadedSize = 0;
-      for (size_t i = 0; i < count-1; ++i)
+      FileReader r(fName);
+      ReaderSource<FileReader> src(r);
+
+      int64_t const readSize = ReadVarInt<int64_t>(src);
+      if (readSize == fileSize)
       {
-        if (m_chunks[i].m_status != CHUNK_COMPLETE)
-          m_chunks[i].m_status = CHUNK_FREE;
-        else
-          downloadedSize += (m_chunks[i+1].m_pos - m_chunks[i].m_pos);
-      }
+        // Load chunks.
+        uint64_t const size = src.Size();
+        int const stSize = sizeof(ChunkT);
+        size_t const count = size / stSize;
+        ASSERT_EQUAL(size, stSize * count, ());
+
+        m_chunks.resize(count);
+        src.Read(&m_chunks[0], stSize * count);
+
+        // Reset status "downloading" to "free".
+        int64_t downloadedSize = 0;
+        for (size_t i = 0; i < count - 1; ++i)
+        {
+          if (m_chunks[i].m_status != CHUNK_COMPLETE)
+            m_chunks[i].m_status = CHUNK_FREE;
+          else
+            downloadedSize += (m_chunks[i + 1].m_pos - m_chunks[i].m_pos);
+        }
 
-      return downloadedSize;
+        return downloadedSize;
+      }
+    }
+    catch (FileReader::Exception const & e)
```

Comment:
`#incldue "coding/reader.hpp"`

Reply:
Why? file_reader is included

Reply:
Yes. `coding/file_reader.hpp` is included. But `FileReader::Exception` is declared in  `coding/reader.hpp`.

Reply:
FileReader inherits its exception from Reader and we're using FileReader::Exception (not Reader::Exception) here.
In any case, reader.hpp is inculded by file_reader.hpp.

Reply:
Oh, yes. Right!